oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -932,17 +932,45 @@ public Map<String, List<String>> getRequestMetadata(URI uri) throws IOException
               + " providing uri to getRequestMetadata.");
     }
 
-    // If scopes are provided but we cannot use self signed JWT, then use scopes to get access
-    // token.
+    if (isDefaultUniverseDomain()) {
+      return getRequestMetadataForGdu(uri);
+    } else {
+      return getRequestMetadataForNonGdu(uri);
+    }
+  }
+
+  private Map<String, List<String>> getRequestMetadataForGdu(URI uri) throws IOException {
+    // If scopes are provided, but we cannot use self-signed JWT or domain-wide delegation is
+    // configured then use scopes to get access token.
     if ((!createScopedRequired() && !useJwtAccessWithScope)
-        || (serviceAccountUser != null && serviceAccountUser.length() > 0)) {
+        || isConfiguredForDomainWideDelegation()) {
       return super.getRequestMetadata(uri);
     }
 
-    // If scopes are provided and self signed JWT can be used, use self signed JWT with scopes.
-    // Otherwise, use self signed JWT with uri as the audience.
+    return getRequestMetadataWithSelfSignedJwt(uri);
+  }
+
+  private Map<String, List<String>> getRequestMetadataForNonGdu(URI uri) throws IOException {
+    // Self Signed JWT is not supported for domain-wide delegation for non-GDU universes
+    if (isConfiguredForDomainWideDelegation()) {
+      throw new IOException(
+          String.format(
+              "Service Account user is configured for the credential. "
+                  + "Domain-wide delegation is not supported in universes different than %s.",
+              Credentials.GOOGLE_DEFAULT_UNIVERSE));
+    }
+
+    return getRequestMetadataWithSelfSignedJwt(uri);
+  }
+
+  /** Provide the access JWT for scopes if provided, for uri as aud otherwise */
+  @VisibleForTesting
+  private Map<String, List<String>> getRequestMetadataWithSelfSignedJwt(URI uri)
+      throws IOException {
+    // If scopes are provided and self-signed JWT can be used, use self-signed JWT with scopes.
+    // Otherwise, use self-signed JWT with uri as the audience.
     JwtCredentials jwtCredentials;
-    if (!createScopedRequired() && useJwtAccessWithScope) {
```

Comment:
is removing `useJwtAccessWithScope` backward compatible?

Reply:
It's a diff making it look confusing ) 

Those lines were moved into private method. You can see that for GDU there is same check (line 945) and for non-gdu it is not needed. 

So no breaking change for GDU